abipy/flowtk/works.py

```diff
@@ -1967,3 +1968,156 @@ def on_all_ok(self):
+
+        Args:
+            multi: a |MultiDataset| object containing a list of 4 datasets or 6 with Kerange.
+                       See abipy/abio/factories.py -> conduc_from_scf_nscf_inputs for details about multi.
+            DDB: a string containing the path to the DDB file.
+            DVDB: a string containing the path to the DVDB file.
+            nbr_proc: Required if withKerange since autoparal doesn't work with optdriver=8.
+            flow: The flow calling the work, only needed if withKerange.
+            withKerange: True if using Kerange.
+            skipInter: Used to skip the interpolation task and compute the fine DVDB during the conductivity task.
+            omp_nbr_thread : Number of omp_thread to use.
+            manager: |TaskManager| of the task. If None, the manager is initialized from the config file.
+        """
+        # Verify Multi
+        if (not withKerange) and (multi.ndtset != 4): #Without kerange, multi should contain 4 datasets
+            raise ValueError("""The |MultiDataset| object does not contain the expected number of dataset.
+                                It should have 4 datasets and it had `%s`. You should generate
+                                multi with the factory function conduc_from_scf_nscf_inputs""" % multi.ndtset)
+        if (withKerange) and (multi.ndtset != 6): # With Kerange, multi should contain 6 datasets
+            raise ValueError("""The |MultiDataset| object does not contain the expected number of dataset.
+                              It should have 6 datasets and it had `%s`.You should generate
+                                multi with the factory function conduc_from_scf_nscf_inputs""" % multi.ndtset)
+        # Make sure both file exists
+        if not os.path.exists(DDB):
+            raise ValueError("The DDB file doesn't exists : `%s`" % DDB)
+        if not os.path.exists(DVDB):
+            raise ValueError("The DVDB file doesn't exists : `%s`" % DVDB)
+        # Verify nbr_proc and flow are defined if withKerange
+        if withKerange and (flow is None or nbr_proc is None):
+            raise ValueError("""When using kerange, the argument flow and nbr_proc must be passed to the function from_filepath
+                                flow = {}, nbr_proc = {}""".format(flow, nbr_proc))
+
+        new = cls(manager=manager)
+
+        new.register_task(multi[0])
+        new.register_task(multi[1], deps={new[0]: "DEN"})
+        taskNumber = 2 # To keep track of the task in new and multi
+
+        if(withKerange): # Using Kerange
+            new.register_task(multi[2], deps={new[1]: "WFK"})
+            new.register_task(multi[3], deps={new[0]: "DEN", new[1]: "WFK", new[2]: "KERANGE.nc"})
+            taskNumber = 4 # We have 2 more task
+
+        if skipInter: # Without separate task for DVDB interpolation
+            new.register_task(multi[taskNumber+1], deps=[Dependency(new[taskNumber-1], "WFK"),
```

Comment:
If recall correctly,

'''python
deps=[Dependency(DDB, "DDB"),  Dependency(DVDB, "DVDB")]
'''

can be replaced by 


```python
deps=[DDB, "DDB"),  DVDB, "DVDB"]
'''